pandas/_libs/lib.pyx

```diff
@@ -2665,6 +2665,8 @@ def maybe_convert_objects(ndarray[object] objects,
             else:
                 seen.object_ = True
                 break
+        elif val is C_NA:
+            seen.object_ = True
```

Comment:
I think missing a `break` after this? See it in all the other branches at least

Reply:
No this is missing on purpose. We don't know which dtype we have with NA and infer-string set, so have to do another pass

Reply:
Ah OK. Does `continue` work here too? Assuming no difference with the current code setup, but continue would be more future-proof in case of a refactor, while also signaling intent

Reply:
Yeah sure